src/backend/marsha/bbb/utils/bbb_utils.py

```diff
@@ -73,41 +82,48 @@ def bbb_api_create(meeting):
         "welcome": meeting.welcome_text,
     }
 
-    api_response = request_bbb_api("create", parameters)
+    api_response = request_api("create", parameters)
     if not api_response.get("message"):
         api_response["message"] = "Meeting created."
+    meeting.started = True
+    meeting.moderator_password = api_response["moderatorPW"]
+    meeting.attendee_password = api_response["attendeePW"]
+    meeting.save()
     return api_response
 
 
-def bbb_api_end(meeting, moderator=False):
-    """Call BBB API to end a meeting."""
-    password = meeting.attendee_password
-    if moderator:
-        password = meeting.moderator_password
+def join(meeting: Meeting, fullname, moderator=False):
+    """Call BBB API to join a meeting."""
     parameters = {
+        "fullName": fullname,
         "meetingID": str(meeting.meeting_id),
-        "password": password,
+        "password": _password(meeting, moderator),
+        "redirect": "true",
     }
-    return request_bbb_api("end", parameters)
+    return request_api("join", parameters, prepare=True)
 
 
-def bbb_api_join(meeting, fullname, moderator=False):
-    """Call BBB API to join a meeting."""
-    password = meeting.attendee_password
-    if moderator:
-        password = meeting.moderator_password
+def end(meeting: Meeting, moderator=False):
+    """Call BBB API to end a meeting."""
     parameters = {
-        "fullName": fullname,
         "meetingID": str(meeting.meeting_id),
-        "password": password,
-        "redirect": "true",
+        "password": _password(meeting, moderator),
     }
-    return request_bbb_api("join", parameters, prepare=True)
+    return request_api("end", parameters)
 
 
-def bbb_api_get_meeting_infos(meeting):
+def get_meeting_infos(meeting: Meeting):
     """Call BBB API to retrieve meeting informations."""
     parameters = {
         "meetingID": meeting.meeting_id,
     }
-    return request_bbb_api("getMeetingInfo", parameters)
+    try:
+        api_response = request_api("getMeetingInfo", parameters)
+        meeting.started = api_response["returncode"] == "SUCCESS"
+        # meeting.started = api_response["running"] == "true"
+        # meeting.save()
```

Comment:
Should not be saved ?

Reply:
:+1:  I fixed related tests accordingly.